### PR TITLE
[mono] Remove gdb xdebug and binary writer support, it hasn't worked in a while.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -4431,21 +4430,6 @@ mini_init (const char *filename, const char *runtime_version)
...
 
-#ifdef XDEBUG_ENABLED
-	char *mono_xdebug = g_getenv ("MONO_XDEBUG");
-	if (mono_xdebug) {
-		mono_xdebug_init (mono_xdebug);
-		g_free (mono_xdebug);
-		/* So methods for multiple domains don't have the same address */
-		mono_dont_free_domains = TRUE;
-		mono_using_xdebug = TRUE;
-	} else if (mini_debug_options.gdb) {
```
Comment (akoeplinger) [inline src/mono/mono/mini/mini-runtime.c:4442]: nit: we might want to print a warning in `mini_parse_debug_option()` if someone tries to set the `gdb` option now.

**Review by akoeplinger — APPROVED** [review]

**Review by lambdageek — APPROVED** [review]

**Comment by akoeplinger** [comment]
This passed on AzDO here: https://dev.azure.com/dnceng/public/_build/results?buildId=1273052&view=results

**Pull request merged**